LB3210.org

```diff
@@ -181,35 +242,143 @@ Full-text search is often important when dealing with data.
   * Service needs to run extra HTTP server
   * Does not compose (multiple metrics backends need to be known on the prometheus side)
   * Advantages: less timing-sensitive
-** Request tracing
+*** Metrics terminology
+- Different frameworks use different terms
+- [[https://micrometer.io/][Micrometer]] uses the following:
+  * /Counter/ (sometimes called /event/): An occurence of a discrete event
+    + e.g. a request coming in
+  * /Gauge/: The size of a single measurable quantity (and its unit)
+    + e.g. the number of active TCP connections
+  * /Timer/: The duration of an activity
+    + e.g. the response time to a request
+  * /Distribution summary/ (sometimes called /histogram/ or even /gauge/): Recorded values (and units) that go with events
+    + e.g. the size of incoming requests in bytes
+** TODO Request tracing
 TODO write about Jaeger and Zipkin
-** Configuration
-- configuration file changes
-  * kubernetes
-  * hot reloading? Not in spring boot (watch file and shutdown instead)
-- environment variables for service injection: don't do it (ordering issues)
-  * use dns instead (e.g. dns-java, akka discovery, [...])
 ** Deployment
+*** Virtualization and containerization
+- First, therer was plain hardware
+- VM abstraction
+  * Decoupling of multiple roles of one server
+  * Memory and disk overhead
+  * Linux optimizations (kernel shared memory)
+- Linux can do many of this natively
+  * /Namespaces/: Hide processes from each other
+  * /Cgroups/: Limit resource usage
+- Containers to make it fast and memory-efficient
+  * Instant startup
 *** Docker
 - Limited to linux in this course
 - Lightweight layer over native cgroups isolation
```

Comment:
maybe you should point out that docker images are typical in 100's of megabytes where other virtualization images (e.g. virtualbox) takes up 10's of gigabytes (feel free to adjust the numbers; the point is that docker images typically is not an issue in modern environments)

Reply:
Yes good one! Also good to go into ubuntu vs. alpine base images.